prover/scripts/pcs_benchmark.rs

```diff
@@ -0,0 +1,229 @@
+
+    println!("Warming up proof...");
+    for _ in 0..3 {
+        full_proof(num_cols, log2_rows);
+    }
+    println!("Starting run of proof...");
+    let now = std::time::Instant::now();
+    for _ in 0..cli.num_runs {
+        full_proof(num_cols, log2_rows);
+    }
+    println!(
+        "Time to generate {} proofs: {:.2}",
+        cli.num_runs,
+        now.elapsed().as_secs_f64()
+    );
+    println!("Warming up pcs_commit...");
+    for _ in 0..3 {
+        pcs_commit(num_cols, log2_rows);
+    }
+    println!("Starting run of pcs_commit...");
+    let now = std::time::Instant::now();
+    for _ in 0..cli.num_runs {
+        pcs_commit(num_cols, log2_rows);
+    }
+    println!(
+        "Time to run {} commits: {:.2}",
+        cli.num_runs,
+        now.elapsed().as_secs_f64()
+    );
+}
+
+fn full_proof(num_cols: usize, log2_rows: usize) {
+    let mut rng = &mut OsRng;
+    let log_degrees = &[log2_rows];
+    let perm = inner_perm();
+    let fri_config = inner_fri_config();
+    let hash = InnerHash::new(perm.clone());
+    let compress = InnerCompress::new(perm.clone());
+    let val_mmcs = InnerValMmcs::new(hash, compress);
+    let dft = InnerDft {};
+    let pcs_val: InnerPcs = InnerPcs::new(
+        log_degrees.iter().copied().max().unwrap(),
+        dft,
+        val_mmcs,
+        fri_config,
+    );
+
+    // Generate proof.
+    let domains_and_polys = log_degrees
+        .iter()
+        .map(|&d| {
+            (
+                <InnerPcs as Pcs<InnerChallenge, InnerChallenger>>::natural_domain_for_degree(
+                    &pcs_val,
+                    1 << d,
+                ),
+                RowMajorMatrix::<InnerVal>::rand(&mut rng, 1 << d, num_cols),
+            )
+        })
+        .sorted_by_key(|(dom, _)| Reverse(dom.log_n))
+        .collect::<Vec<_>>();
+    let (commit, data) = <InnerPcs as Pcs<InnerChallenge, InnerChallenger>>::commit(
+        &pcs_val,
+        domains_and_polys.clone(),
+    );
+    let mut challenger = InnerChallenger::new(perm.clone());
+    challenger.observe(commit);
+    let zeta = challenger.sample_ext_element::<InnerChallenge>();
+    let points = domains_and_polys
+        .iter()
+        .map(|_| vec![zeta])
+        .collect::<Vec<_>>();
+    let (opening, proof) = pcs_val.open(vec![(&data, points)], &mut challenger);
+
+    // Verify proof.
+    let mut challenger = InnerChallenger::new(perm.clone());
+    challenger.observe(commit);
+    challenger.sample_ext_element::<InnerChallenge>();
+    let os: Vec<(
+        TwoAdicMultiplicativeCoset<InnerVal>,
+        Vec<(InnerChallenge, Vec<InnerChallenge>)>,
+    )> = domains_and_polys
+        .iter()
+        .zip(&opening[0])
+        .map(|((domain, _), mat_openings)| (*domain, vec![(zeta, mat_openings[0].clone())]))
+        .collect();
+    pcs_val
+        .verify(vec![(commit, os.clone())], &proof, &mut challenger)
+        .unwrap();
+
+    // Test the recursive Pcs.
```

Comment:
Unecessary code, don't need to test this stuff! This is running it inside the program.